integration/distinct_compat_test.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+
+	"github.com/FerretDB/FerretDB/integration/setup"
+	"github.com/FerretDB/FerretDB/integration/shareddata"
+)
+
+// distinctCompatTestCase describes count compatibility test case.
+type distinctCompatTestCase struct {
+	field         string                   // required
+	skipForTigris string                   // optional
+	filter        bson.D                   // required
+	resultType    compatTestCaseResultType // defaults to nonEmptyResult
+}
+
+func testDistinctCompat(t *testing.T, testCases map[string]distinctCompatTestCase) {
+	t.Helper()
+
+	// Use shared setup because distinct queries can't modify data.
+	// TODO Use read-only user. https://github.com/FerretDB/FerretDB/issues/1025
+	s := setup.SetupCompatWithOpts(t, &setup.SetupCompatOpts{
+		Providers:                shareddata.AllProviders(),
+		AddNonExistentCollection: true,
+	})
+	ctx, targetCollections, compatCollections := s.Ctx, s.TargetCollections, s.CompatCollections
+
+	for name, tc := range testCases {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Helper()
+
+			if tc.skipForTigris != "" {
+				setup.SkipForTigrisWithReason(t, tc.skipForTigris)
+			}
+
+			t.Parallel()
+
+			filter := tc.filter
+			require.NotNil(t, filter, "filter should be set")
+
+			var nonEmptyResults bool
+			for i := range targetCollections {
+				targetCollection := targetCollections[i]
+				compatCollection := compatCollections[i]
+				t.Run(targetCollection.Name(), func(t *testing.T) {
+					t.Helper()
+
+					targetRes, targetErr := targetCollection.Distinct(ctx, tc.field, tc.filter)
+					compatRes, compatErr := compatCollection.Distinct(ctx, tc.field, tc.filter)
```

Comment:
I know it's not mongosh, but still I assume it's mongo helper function. Are we sure that we want use it and not `runCommand`? 

From https://www.mongodb.com/docs/manual/reference/command/distinct/:
> In [mongosh](https://www.mongodb.com/docs/mongodb-shell/#std-program-mongosh), this command can also be run through the [db.collection.distinct()](https://www.mongodb.com/docs/manual/reference/method/db.collection.distinct/#mongodb-method-db.collection.distinct) helper method..
> 
> Helper methods are convenient for [mongosh](https://www.mongodb.com/docs/mongodb-shell/#std-program-mongosh) users, but they may not return the same level of information as database commands. In cases where the convenience is not needed or the additional return fields are required, use the database command.

Reply:
I think we had that question before, but let's discuss it again :) 

The idea is that we test compatibility with driver first. We don't care about monogsh, but we do care about mongo driver for Go. So, we use `Distinct` function because mongo driver offers it. For example, this is what we also did when we tested ordered inserts - we used `insertOne` and `insertMany` for main tests. It helped us reveal that we used to return a wrong response type. 

Imagine, you are a developer and you want to write an app that would use FerretDB and you need to call distinct. Most likely, as a developer, you will call `targetCollection.Distinct`, not `RunCommand`. This is why we want to test such cases first.

We will truly need to test with `runCommand`, but later. For example, we will need to cover cases when `key` is an array, or some other weird data is passed. We don't focus on it now because it'll take way too much time.

Reply:
You right! Yeah, let's focus on the drivers first. We could create an issue or todo for runCommand later!